pkg/resource/deploy/step_generator.go

```diff
@@ -421,23 +424,57 @@ func (sg *stepGenerator) inheritedChildAlias(
 		aliasName)
 }
 
+// configureGoal mutates the passed goal to inherit the correct ResourceOptions from its parent.
+func (sg *stepGenerator) configureGoal(goal *resource.Goal) result.Result {
+	originalParent := goal.Parent
+
+	if originalParent != "" {
+		parentGoal, ok := sg.resourceGoals[originalParent]
+		if !ok {
+			// Check if originalParent is an Alias
+			parentURN, ok := sg.aliases[originalParent]
```

Comment:
Parents won't ever be aliases, you can just take the result of checkParent to know the parent URN (or "")